/*
 * Copyright (c) 2020-2020, Exactpro (Exactpro Systems Limited)
 * www.exactpro.com
 * Build Software to Test Software
 *
 * All rights reserved.
 * This is unpublished, licensed software, confidential and proprietary 
 * information which is the property of Exactpro Systems Limited or its licensors.
 */

syntax = "proto3";
package th2;

import "common.proto";
import "win_act.proto";
import "order_ticket.proto";
import "google/protobuf/timestamp.proto";
import "google/protobuf/wrappers.proto";

option java_multiple_files = true;
option java_package = "com.exactpro.th2.act.grpc.fe";

service OrderBookService {
  rpc amendOrder(ModifyOrderDetails) returns (th2.ActResponse) {}
  rpc splitOrder(ModifyOrderDetails) returns (th2.ActResponse) {}
  rpc splitLimit(ModifyOrderDetails) returns (th2.ActResponse) {}
  rpc cancelOrder(CancelOrderDetails) returns (th2.ActResponse) {}
  rpc getOrdersDetails(GetOrdersDetailsRequest) returns (th2.ActResponse) {}
}

message ModifyOrderDetails {
  BaseRequest base = 1;
  map<string, string> filter = 2;
  OrderDetails orderDetails = 3;
}

message CancelOrderDetails {
  BaseRequest base = 1;
  map<string, string> filter = 2;
  google.protobuf.BoolValue cancelChildren = 3;
  string comment = 4;
}

message GetOrdersDetailsRequest {
  BaseRequest base = 1;
  string extractionId = 2;
  OrdersDetailsInfo orderDetails = 3;
}

message OrdersDetailsInfo {
  map<string, string> filter = 1;
  bool extractCount = 2;
  string countId = 3;
  repeated ExtractionInfo extractionInfo = 4;
}

message ExtractionInfo {
  int32 number = 2;
  repeated ExtractionFieldsDetails orderDetails = 3;
  OrdersDetailsInfo subOrders = 4;
}
